src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java

```diff
@@ -171,15 +175,32 @@ private void computeCodeActionsForSonarLintIssues(Diagnostic diagnostic, List<Ei
         newCodeAction.setCommand(new Command(fix.message(), SONARLINT_QUICK_FIX_APPLIED, List.of(ruleKey)));
         codeActions.add(Either.forRight(newCodeAction));
       });
+
+      if (hasBinding && versionedIssue.getIssue() instanceof DelegatingIssue) {
+        var serverIssueKey = ((DelegatingIssue) versionedIssue.getIssue()).getServerIssueKey();
+        var muteIssueAction = createMuteIssueCodeAction(diagnostic, ruleKey, serverIssueKey, uri, false);
+        codeActions.add(Either.forRight(muteIssueAction));
+      }
     }
     addRuleDescriptionCodeAction(params, codeActions, diagnostic, ruleKey, ruleContextKey);
     issueForDiagnostic.ifPresent(versionedIssue -> addShowAllLocationsCodeAction(versionedIssue, codeActions, diagnostic, ruleKey, isNotebookCellUri));
-    if (binding.isEmpty()) {
+    if (!hasBinding) {
       var titleDeactivate = String.format("Deactivate rule '%s'", ruleKey);
       codeActions.add(newQuickFix(diagnostic, titleDeactivate, SONARLINT_DEACTIVATE_RULE_COMMAND, List.of(ruleKey)));
     }
   }
 
+  @NotNull
+  private CodeAction createMuteIssueCodeAction(Diagnostic diagnostic, String ruleKey, String serverIssueKey,  URI fileUri, boolean isTaintIssue) {
+    var workspace = workspaceFoldersManager.findFolderForFile(fileUri).orElseThrow(() -> new IllegalStateException("No workspace found"));
+    var workspaceUri = workspace.getUri();
+    var muteIssueAction = new CodeAction(String.format(SONARLINT_ACTION_PREFIX + "Mute this issue violating rule '%s'", ruleKey));
+    muteIssueAction.setKind(CodeActionKind.QuickFix);
+    muteIssueAction.setDiagnostics(List.of(diagnostic));
+    muteIssueAction.setCommand(new Command("Mute this issue", MUTE_ISSUE, List.of(workspaceUri.toString(), serverIssueKey, fileUri, isTaintIssue)));
```

Comment:
Sorry to come back with another comment but please have a look at[ this thread](https://sonarsource.slack.com/archives/CFX1J97GT/p1685696508153309). As I understand, we should move away from `mute` and use `resolve` terminology everywhere.

Reply:
That's good point 👍 Addressed in the last commit